dvc/utils/fs.py

```diff
@@ -21,12 +21,12 @@ def get_inode(path):
 
 
 def get_mtime_and_size(path, dvcignore):
-    if os.path.isdir(path):
+    if os.path.isdir(fspath_py35(path)):
         size = 0
         files_mtimes = {}
         for file_path in walk_files(path, dvcignore):
             try:
-                stat = os.stat(file_path)
+                stat = os.stat(fspath_py35(file_path))
```

Comment:
The `fspath_py35()` wrapper should not be needed here.

Reply:
@Suor `os.stat` takes both types?

Reply:
`file_path` yielded from `walk_files()` is always a string.